### DISCUSSION TITLE
Simplify the use case food-packaging

### DISCUSSION THREAD

**Discussion by kentbill** [TimefoldAI/timefold-solver#470] [General]
Why do we need 3 shadow variables in the food packaging example: startCleaningDateTime, startProductionDateTime, and endDateTime? Can we only define startCleaningDateTime as a shadow variable, and the other 2 variables deduced from it:

1. The **startProductionDateTime** is startCleaningDateTime plus cleaningDuration.
2. The **endTime** is startProductionDateTime plus the duration of the job.

Did I overlook some situations that have to let startCleaningDateTime  and startProductionDateTime be shadow variables?

**Comment by ge0ffrey** [comment]
That would work too. It's a trade-off.

I think it wrote it this way to make it easier to reuse those values in the UI, without having to calculate them there... Not a good reason :)

Or to avoid the hashmap lookup to calculate the cleaning duration, when endTime is used in the constraint(s). Not sure if that matters much though.

**Reply by kentbill** [reply]
got it, thank you Geoffrey!
